examples/step-60/step-60.cc

```diff
@@ -0,0 +1,1077 @@
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the deal.II distribution.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * Authors: Luca Heltai, Giovanni Alzetta,
+ * International School for Advanced Studies, Trieste, 2018
+ */
+
+// @sect3{Include files}
+// Most of these have been introduced elsewhere, we'll comment only on the new ones.
+
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/timer.h>
+
+// The parameter acceptor class is the first novelty of this tutorial program:
+// in general parameter files are used to steer the execution of a program
+// at run time. While even a simple approach saves compiling time, as the same
+// executable can be run with different parameter settings, it can become
+// difficult to handle hundreds of parameters simultaneously while maintaining
+// compatibility between different programs. This is where the class
+// ParameterAcceptor proves useful.
+//
+// This class is used to define a public interface for classes that want to use
+// a single global ParameterHandler to handle parameters. The class provides a
+// static ParameterHandler member, namely ParameterAcceptor::prm, and
+// implements the "Command design patter" (see, for example, E. Gamma, R. Helm,
+// R. Johnson, J. Vlissides, Design Patterns: Elements of Reusable
+// Object-Oriented Software, Addison-Wesley Professional, 1994.
+// https://goo.gl/FNYByc).
+//
+// ParameterAcceptor provides a global subscription mechanism. Whenever an
+// object of a class derived from ParameterAcceptor is constructed, a pointer
+// to that object-of-derived-type is registered, together with a section entry
+// in the parameter file. Such registry is traversed upon invocation of the
+// single function ParameterAcceptor::initialize(file.prm) which in turn makes
+// sure that all classes stored in the global register declare the parameters
+// they will be using, and after having declared them, it reads the content of
+// `file.prm` to parse the actual parameters.
+//
+// If you call the method ParameterHandler::add_parameter for each of the
+// parameters you want to use in your code, there is nothing else you need to
+// do. If you are using an already existing class that provides the two
+// functions `declare_parameters` and `parse_parameters`, you can still use
+// ParameterAcceptor, by encapsulating the existing class into a
+// ParameterAcceptorProxy class.
+//
+// In this example, we'll use both strategies, using ParameterAcceptorProxy for
+// deal.II classes, and deriving our own parameter classes directly from
+// ParameterAcceptor.
+
+#include <deal.II/base/parameter_acceptor.h>
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/grid/grid_tools_cache.h>
+
+// The other new include file is the one that contains the GridTools::Cache
```

Comment:
Please move this discussion *above* the `#include` above, or make it clear that the discussion refers to the previous include file.